src/connectors/index.ts

```diff
@@ -1,65 +1,107 @@
-import getLibrary from '../utils/getLibrary'
-import { CustomNetworkConnector } from './CustomNetworkConnector'
-import { CustomWalletConnectConnector } from './CustomWalletConnectConnector'
-import { CustomWalletLinkConnector } from './CustomWalletLinkConnector'
-
-export const INFURA_PROJECT_ID = '0ebf4dd05d6740f482938b8a80860d13'
-
-export const network = new CustomNetworkConnector({
-  urls: {
-    [ChainId.MAINNET]: `https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`,
-    [ChainId.XDAI]: 'https://rpc.gnosischain.com/',
-    [ChainId.ARBITRUM_ONE]: 'https://arb1.arbitrum.io/rpc',
-    [ChainId.POLYGON]: 'https://polygon-rpc.com/',
-  },
-  defaultChainId: ChainId.MAINNET,
-})
-
-export const injected = new InjectedConnector({
-  supportedChainIds: [
-    ChainId.MAINNET,
-    ChainId.RINKEBY,
-    ChainId.ARBITRUM_ONE,
-    ChainId.ARBITRUM_RINKEBY,
-    ChainId.XDAI,
-    ChainId.POLYGON,
-  ],
-})
-
-// mainnet only
-export const walletConnect = new CustomWalletConnectConnector({
-  rpc: {
-    [ChainId.XDAI]: 'https://rpc.gnosischain.com/',
-    [ChainId.MAINNET]: `https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`,
-  },
-  bridge: 'https://bridge.walletconnect.org',
-  qrcode: true,
-  pollingInterval: 15000,
-})
-
-let networkLibrary: providers.Web3Provider | undefined
-export function getNetworkLibrary(): providers.Web3Provider {
-  return (networkLibrary = networkLibrary ?? getLibrary(network.provider))
+import { CoinbaseWallet } from '@web3-react/coinbase-wallet'
+import { initializeConnector, Web3ReactHooks } from '@web3-react/core'
+import { MetaMask } from '@web3-react/metamask'
+import { Network } from '@web3-react/network'
+import { Connector } from '@web3-react/types'
+import { WalletConnect } from '@web3-react/walletconnect'
+
+import SWAPR_LOGO from './../assets/svg/swapr.svg'
+import { ConnectorType, RPC_URLS } from './../constants'
+
+export interface Connection {
+  connector: Connector
+  hooks: Web3ReactHooks
+  type: ConnectorType
+}
+
+function onError(error: Error) {
+  console.debug(`web3-react error: ${error}`)
+}
+
+// Network
+export const [web3Network, web3NetworkHooks] = initializeConnector<Network>(
+  actions =>
+    new Network({
+      actions,
+      urlMap: RPC_URLS,
+      defaultChainId: ChainId.MAINNET,
+    })
+)
+
+export const networkConnection: Connection = {
+  connector: web3Network,
+  hooks: web3NetworkHooks,
+  type: ConnectorType.NETWORK,
 }
 
-// walletLink implements Metamask's RPC and should respond to most it's methods: window.ethereum.isMetaMask === true
-// More info: https://github.com/walletlink/walletlink
-export const walletLink = new CustomWalletLinkConnector({
-  url: `https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`,
-  appName: 'Swapr',
-  appLogoUrl: swprLogo,
-  supportedChainIds: [
-    ChainId.MAINNET,
-    ChainId.RINKEBY,
-    ChainId.ARBITRUM_ONE,
-    ChainId.ARBITRUM_RINKEBY,
-    ChainId.XDAI,
-    ChainId.POLYGON,
-  ],
-})
+// TODO is Injected needed?
```

Comment:
should we remove these comments?

Reply:
Removed [a5d92a4](https://github.com/levelkdev/swapr-dapp/pull/1330/commits/a5d92a4255550303e4b0586fb90a05b0f4344d52)